### ISSUE TITLE
Markdown and Doxygen in Code Blocks

### ISSUE THREAD

**Issue by HaoZeke** [doxygen/doxygen#8414]
**Describe the bug**
Currently putting doxygen code blocks in markdown code doesn't get parsed with the INLINE_SOURCES option on.

Sample code for replication:

```cpp
/**
 * @note something
 */
int main(int bug);
```

**Expected behavior**
I believe these code blocks should be kept as is. The only workaround currently is to use an unsupported syntax highlighting language but that means that the code is ugly.

**Version**
1.9.1 on ArchLinux

**Comment by albert-github** [comment]
I don't understand what you see / want to accomplish and what you would like to see, please add some images to the issue that shine a light on it.

It is also unclear which settings you have used.
So can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by HaoZeke** [comment]
@albert-github, sorry I wasn't clear before. I have updated this with a minimum example and screenshots.

**Comment by albert-github** [comment]
I think your problem is the setting of `STRIP_CODE_COMMENTS`, from the documentation:
> Setting the STRIP_CODE_COMMENTS tag to YES will instruct doxygen to hide any special comment blocks from generated source code fragments. Normal C, C++ and Fortran comments will always remain visible.
> The default value is: YES.

I think setting `STRIP_CODE_COMMENTS=NO` will result in the required output.

**Comment by HaoZeke** [comment]
Yes but then it'll show up in the documentation with `INLINE_SOURCES` which is undesirable. The desired output is to have `STRIP_CODE_COMMENTS=NO` in `.md` files while still being on for `cpp` and `h` files.

**Comment by albert-github** [comment]
OK I do now what you try to accomplish, in short it is the ability to disable the `STRIP_CODE_COMMENTS` for certain file types or another possibility would be disable the `STRIP_CODE_COMMENTS` to for certain comment blocks. This is a difficult problem.

A also not nice workaround would be:
~~~
```unparsed
/**
 * @details This is completely useless
 * @note Does not render
 * */
```
```cpp
int main(int num) {
    if(num==1){
        return 0;
    }
    else {
        return 1;
    }
}
```
~~~

but here we get a (small) empty space between the 2 blocks.

- maybe some special "language", e.g. `cpp[nostrip]` or `[nostrip]cpp` should be introduced for this. @doxygen what is your opinion?

**Comment by jrsteensen** [comment]
I ran into this today as well and was just starting to look for solutions. Starting a repository using doxygen for docs, and was working on the markdown formatted styleguide. Doyxgen stripped out all the comments in the ````c` formatted code blocks. For now, I'm using unformatted blocks to show the doxygen comments in there, but would prefer to show formatted code blocks with the doxygen-formatted code blocks still showing the leading and trailing doxygen comments.

**Comment by albert-github** [comment]
@jrsteensen 

- What do you mean with:
   > I'm using unformatted blocks to show the doxygen comments in there, but would prefer to show formatted code blocks with the doxygen-formatted code blocks still showing the leading and trailing doxygen comments.
- how should the result look like?
- which doxygen version did you use?

All in all:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by jrsteensen** [comment]
@albert-github

Sample from styleguide.md: 

* Comment every variable or constant declaration with a description of what the variable does. Use trailing doxygen compatible comments unless it is complex enough to require a leading/trailing block comment. You may use markdown compatible comments within the doxygen comments.
    ```c
    /**
    * @brief Pilots may want the launch bar to automatically release when the throttles advance to MIL power.
    * The sim's version of the F/A-18 does *NOT* have this feature.  The define allows setting it as desired, default is *false*.
    *
    */
    #define LBAR_SW_AUTORETRACT false

    // Define pins for DCS-BIOS per interconnect diagram.
    #define HOOKB_SW A0 ///< Hook Bypass Switch
    #define LBAR_SW A1 ///< Launch Bar Switch
    #define FLAPS_SW2 A2 ///< Flaps Switch Position 2
    #define FLAPS_SW1 A3 ///< Flaps Switch Position 1
    #define LBAR_RET 2    ///< Launch Bar Switch Mag

    //Declare variables for custom non-DCSBios logic for mag-switches
    bool hookLeverState = LOW;        ///< Initializing for correct cold/ground start position as switch releases on power off.
    unsigned long hookLeverTime = 0;  ///< Initializing variable to hold time of last hook change.
    bool wowLeft = true;              ///< Initializing weight-on-wheel value for cold/ground start.
    ```

 When the above is parsed via doxygen, it strips out the @brief comment and trailing doxygen comments (when I'm trying to show an example of how to use them.)
 
 IS: 
![image](https://github.com/doxygen/doxygen/assets/2449496/f31eb4b7-bc9e-44a3-99f3-a3fab4821040)

DESIRED: (as rendered in cLion)
![image](https://github.com/doxygen/doxygen/assets/2449496/f5c94a7f-8ab1-4aa9-98ea-bb7dee27bb6f)

If I remove the `c` tag from the comment block, it shows it unformatted (i.e. no color markup), but displays the doxygen comments:
![image](https://github.com/doxygen/doxygen/assets/2449496/0f009cbb-ef72-4d04-bcd0-5a5c378f2055)


### Applicable Files
[STYLEGUIDE.md](https://github.com/doxygen/doxygen/files/14486086/STYLEGUIDE.md)
[Doxyfile.zip](https://github.com/doxygen/doxygen/files/14486101/Doxyfile.zip) (Zipped for GitHub upload compatibility)

**Comment by albert-github** [comment]
- the images are not 100% readable, especially the clion one) as normally when clicking (in GitHub) on it I get the image, but now I get something starting with `https://private-user-images.githubusercontent.com`
- regarding the doxygen comments, in source code they are normally stripped but when using the setting `STRIP_CODE_COMMENTS=NO` this won't happen.
